### PR TITLE
fixed issue for multi-line code for HF model and fixed Capitalized Integer dtypes Failure

### LINKED ISSUE #327: Capitalized Integer dtypes Fail
### 🐛 Describe the bug

Looks like every capitialized integer dtype (supporting NA) fails (invalid dtype) while uncapitalized works. Using v0.6.7. I was unable to find a list of (un)supported dtypes so I assume this is unexpected behavior.

    df = pd.DataFrame([0,1],columns=['col'],dtype='Int64') # fails but 'int64' works!
    print(type(df),df)
    pandas_ai.clear_cache()
    pandas_ai(df, prompt='Show dataframe') # using openai

### LINKED ISSUE #365: No output for multi-line code for HF model
### 🐛 Describe the bug

I ran this example from _getting started.ipynb_ to plot a chart of the GDP of the countries using the HF Starcoder model. 
The model generated the multi-line code but only one line got executed 

```# Import basis libraries
import pandas as pd
from pandasai import PandasAI
from pandasai.llm.starcoder import Starcoder
API_KEY = 'HF model key' #@param {type:"string"}
df = pd.DataFrame({
    "country": ["United States", "United Kingdom", "France", "Germany", "Italy", "Spain", "Canada", "Australia", "Japan", "China"],
    "gdp": [21400000, 2940000, 2830000, 3870000, 2160000, 1350000, 1780000, 1320000, 516000, 14000000],
    "happiness_index": [7.3, 7.2, 6.5, 7.0, 6.0, 6.3, 7.3, 7.3, 5.9, 5.0]
})
# Model Initialisation
llm = Starcoder(api_token=API_KEY)
pandas_ai = PandasAI(llm, conversational=False, verbose=True)
# Enter Prompt related to data or Select from Pre-defined for demo purposes.
prompt = 'Plot the histogram of countries showing for each the gdp, using different colors for each bar' #@param [ "What is the relation between GDP and Happines Index", "Plot the histogram of countries showing for each the gpd, using different colors for each bar", "GDP of Top 5 Happiest Countries?"]  {allow-input: true}
response = pandas_ai.run(df, prompt=prompt,is_conversational_answer=False)
```

**output:** 

2023-07-14 21:03:28 [INFO] Question: Plot the histogram of countries showing for each the gdp, using different colors for each bar
2023-07-14 21:03:28 [INFO] Running PandasAI with starcoder LLM...
2023-07-14 21:03:28 [INFO] Prompt ID: 12a91ba4-eb49-4d54-a1ca-01d30e8cdc30
2023-07-14 21:03:36 [INFO] 
                        Code generated:
                        ```
                        # Import the necessary librariesimport matplotlib.pyplot as pltimport seaborn as sns# Plot the histogram of countries showing for each the gdp, using different colors for each barsns.set_style("darkgrid")sns.set_palette("pastel")sns.barplot(x="gdp", y="country", data=df)plt.show()
                        ```
                    
2023-07-14 21:03:36 [INFO] 
Code running:
```
plt.close('all')
```
2023-07-14 21:03:44 [INFO] Answer: 
2023-07-14 21:03:44 [INFO] Executed in: 16.442320108413696s

### PR DESCRIPTION
**Pull request by kunal588**
Fixed issue for multi-line code output when using HF model. Made changes in hugging face class of base.py inside LLM.
Fixed Capitalized Integer dtypes Failure

- [x] closes #365 
- [x] closes #327 
- [x] All [code checks passed](https://github.com/gventuri/pandas-ai#contributing).

### CODE REVIEW

File: pandasai/helpers/anonymizer.py [hunk 1]
```diff
@@ -175,5 +178,12 @@ def anonymize_dataframe_head(
                 pd.eval(cell_value) if cell_value in ["True", "False"] else cell_value
             )
     # restore the original data types
-    data_frame = data_frame.astype(dtypes)
+    for i in range(len(data_frame.columns)):
+        dt = data_frame.dtypes[i]
+        col = data_frame.columns[i]
+        if dt == "Int64":
+            data_frame[col] = data_frame[col].astype("float64")
```
Comment (gventuri) [inline pandasai/helpers/anonymizer.py:185]: Why do you convert to `float64` to then convert to `int64`?

File: pandasai/helpers/anonymizer.py [hunk 2]
```diff
@@ -175,5 +178,12 @@ def anonymize_dataframe_head(
                 pd.eval(cell_value) if cell_value in ["True", "False"] else cell_value
             )
     # restore the original data types
-    data_frame = data_frame.astype(dtypes)
+    for i in range(len(data_frame.columns)):
```
Comment (gventuri) [inline pandasai/helpers/anonymizer.py:181]: Could you add a comment to make it more understandable? This anonymizer is growing in complexity, we'd better have it well specced!

Comment (gventuri) [inline pandasai/helpers/anonymizer.py:181]: Also, do you think we can have a unit test for this?

**Review by gventuri — COMMENTED** [review]
Well done, only a couple of minor changes!

File: pandasai/helpers/anonymizer.py (hunk 1 above)
Comment (kunal588) [reply pandasai/helpers/anonymizer.py:185]: actually, as in the code, we are converting the data frame to string, but when the data frame is Int64 type and in the anonymizer algorithm converting the string back to its original Int64 type is not doable. To convert the string back to Int64 type will need to first convert it to float64 and then Int64. (bec. float64 handles null values(np.nan) as well).

File: pandasai/helpers/anonymizer.py (hunk 2 above)
Comment (kunal588) [reply pandasai/helpers/anonymizer.py:181]: ok, I will add that

Comment (kunal588) [reply pandasai/helpers/anonymizer.py:186]: I have added the comments

Comment (kunal588) [reply pandasai/helpers/anonymizer.py:186]: Added one unit test

File: pandasai/helpers/anonymizer.py (hunk 1 above)
Comment (gventuri) [reply pandasai/helpers/anonymizer.py]: Great point, I missed it! Thanks a lot for clarifying!

Comment (alikhanak123) [reply pandasai/helpers/anonymizer.py]: I am facing the same issue, and I don't know even after using the above solution, I still get the error, Int64 not understood

Comment (alikhanak123) [reply pandasai/helpers/anonymizer.py]: image

**Pull request merged**